midp/midp.js

```diff
@@ -1049,47 +1035,45 @@ Native["com/sun/midp/events/EventQueue.getNativeEventQueueHandle.()I"] = functio
 };
 
 Native["com/sun/midp/events/EventQueue.resetNativeEventQueue.()V"] = function() {
-    console.warn("EventQueue.resetNativeEventQueue.()V not implemented");
+    MIDP.nativeEventQueues[$.ctx.runtime.isolate.id] = [];
 };
 
 Native["com/sun/midp/events/EventQueue.sendNativeEventToIsolate.(Lcom/sun/midp/events/NativeEvent;I)V"] =
 function(obj, isolateId) {
-    if (!MIDP.nativeEventQueues[isolateId]) {
-      MIDP.nativeEventQueues[isolateId] = [];
-    }
+    var e = { type: obj.type };
+    obj.klass.classInfo.fields.forEach(function(field) {
+        e[field.name] = field.get(obj);
+    });
```

Comment:
This shouldn't matter much because sendNativeEventToIsolate is used just a couple of times (to start a MIDlet and to set a MIDlet as the foreground MIDlet), but I've done it anyway in 20416a5.